[Lim Zhan Feng] iP

LolfoollorS opened the pull request:
# Clevernotbot
![CleverNotBotLogo](https://user-images.githubusercontent.com/110681493/189236848-a4ecc755-1493-4e21-8cca-7ddece77386a.png)
>“Humans Need Not Apply.” – CPG Grey ([source](https://www.youtube.com/watch?v=7Pq-S557XQU))

Trying to relive the old days? Well look no further. Try Clevernotbot! It's
- has a task tracking capability
- text-based
- easy to learn

Did I also mention that it is **FREE OF CHARGE**? All you need to do now is,

1. download it from [here](https://github.com/LolfoollorS/ip/releases/tag/A-Jar).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

Features:

- [x] Managing tasks
- [x] Managing deadlines
- [ ] ~~Reminders~~ (pending)
If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

LolfoollorS committed:
Added level 1- 6, A-Collections

LolfoollorS committed:
Remove Text block literal (Not supported in JDK11)

LolfoollorS committed:
Add Level 7

LolfoollorS committed:
Add Level-8

LolfoollorS committed:
Merge branch 'branch-Level-7'

LolfoollorS committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java

LolfoollorS committed:
Update AddFunctionDeadLine.java
Update Deadline.java

LolfoollorS committed:
Merge remote-tracking branch 'origin/add-gradle-support'

LolfoollorS committed:
Add A-MoreOOP
Update all functions to command.
Update Handler to Parser
Update UITextbox to UI

LolfoollorS committed:
Add packages to all the relevant classes

LolfoollorS committed:
Add A-JUnit: ParserTest, TaskListTest, ToDoTest

LolfoollorS committed:
Add A-Jar

LolfoollorS committed:
Add JavaDocs for all classes except CleverNotBot.java .

LolfoollorS committed:
Revert "Add JavaDocs for all classes except CleverNotBot.java ."

This reverts commit 189448e3c8033f5f58581fa481a14982c8b068a6.

LolfoollorS committed:
Add JavaDocs to all non main classes.

LolfoollorS committed:
Add Coding Standard to code.

LolfoollorS committed:
Add Level 9

LolfoollorS committed:
Merge branch 'branch-A-JavaDoc'

LolfoollorS committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/clevernotbot/AddDeadLineCommand.java
#	src/main/java/clevernotbot/AddEventCommand.java
#	src/main/java/clevernotbot/AddToDoCommand.java
#	src/main/java/clevernotbot/ByeCommand.java
#	src/main/java/clevernotbot/CleverNotBotException.java
#	src/main/java/clevernotbot/Deadline.java
#	src/main/java/clevernotbot/DefaultCommand.java
#	src/main/java/clevernotbot/DeleteCommand.java
#	src/main/java/clevernotbot/Event.java
#	src/main/java/clevernotbot/GreetCommand.java
#	src/main/java/clevernotbot/ListCommand.java
#	src/main/java/clevernotbot/MarkCommand.java
#	src/main/java/clevernotbot/Parser.java
#	src/main/java/clevernotbot/Storage.java
#	src/main/java/clevernotbot/Task.java
#	src/main/java/clevernotbot/TaskList.java
#	src/main/java/clevernotbot/ToDo.java
#	src/main/java/clevernotbot/UI.java
#	src/main/java/clevernotbot/UnmarkCommand.java

LolfoollorS committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/clevernotbot/TaskList.java

LolfoollorS committed:
Update MarkCommand.java
Update UnmarkCommand.java

ShawnChew commented on src/main/java/clevernotbot/Command.java:
```diff
@@ -0,0 +1,48 @@
+ */
+public abstract class Command {
+    private String commandName;
+    private boolean exit;
```
Perhaps you could rename the variable to make it sound like a boolean.

ShawnChew commented on src/main/java/clevernotbot/Deadline.java:
```diff
@@ -0,0 +1,118 @@
+     * @return A toggled version of deadline. (Completed = true -> Completed = false)
+     */
+    @Override
+    public Task toggleCompleted() {
```
Name of method sounds a little misleading, perhaps you could name it as toggleCompletionStatus.

ShawnChew commented on src/main/java/clevernotbot/DeleteCommand.java:
```diff
@@ -0,0 +1,45 @@
+            tasks.removeTask(deletedTask);
+            storage.writeToFile(tasks.getTaskList());
+            textBox.chat(String.format(
+                    "Noted. I've removed this task:" +
+                            "\n  %s" +
+                            "\nNow you have %d tasks in the list."
```
Line break should appear before the operator.

ShawnChew commented on src/main/java/clevernotbot/AddDeadLineCommand.java:
```diff
@@ -0,0 +1,60 @@
+    /**
+     * Runs the add deadline command.
+     *
+     * @param tasks   The task list used to store all tasks.
+     * @param textBox UI for the textbox.
+     * @param storage The data where it is stored.
```
I like that you aligned the descriptions for the params to make it look neater.

ShawnChew commented on src/main/java/clevernotbot/Event.java:
```diff
@@ -0,0 +1,61 @@
+    }
+
+    /**
+     * Toggles the completion of event task.
+     *
+     * @return A toggled version of event. (Completed = true -> Completed = false)
```
Perhaps you could write the content in the parenthesis, in the second sentence of the description instead of on the @return line.

ShawnChew commented on src/main/java/clevernotbot/Storage.java:
```diff
@@ -0,0 +1,113 @@
+        System.out.println(file.getAbsolutePath());
+        System.out.println(String.join(",",content));
+         */
+        //"1".equals(content[1]) is to convert it to boolean
```
I like how you commented on what the code is doing.

ShawnChew commented on src/main/java/clevernotbot/ToDo.java:
```diff
@@ -0,0 +1,57 @@
+        return "T";
+    }
+
+    /**
+     * Toggles the completion of todo task.
+     *
+     * @return A toggled version of todo. (Completed = true -> Completed = false)
+     */
```
You can omit header comments for overriding methods (provided the parent method's Javadoc applies exactly as is to the overridden method).

ShawnChew reviewed (commented):
Overall I think you followed the Java Coding Standard closely. 👍

PokezardVGC commented on src/main/java/clevernotbot/AddEventCommand.java:
```diff
@@ -0,0 +1,54 @@
+     * Constructor for the AddEventCommand.
+     *
+     * @param commandName Description of Command.
+     * @param exit        Checking if program intends to exit.
```
Good use of alignment to keep description on one side and parameters on the other

PokezardVGC commented on src/main/java/clevernotbot/Event.java:
```diff
@@ -0,0 +1,61 @@
+ * a task with a time range e.g., <code>Aug 6th 2-4pm</code>
+ */
+public class Event extends Task {
+    private String at;
```
possible to make this private final?

PokezardVGC commented on src/main/java/clevernotbot/TaskList.java:
```diff
@@ -0,0 +1,110 @@
+ */
+public class TaskList {
+    private ArrayList<Task> tasks;
+    private int size;
```
likewise, possible to make it private final? Maybe create a new TaskList object with each function call

LolfoollorS committed:
Add catch for non numeric number after command such as `mark a` `unmark a` `delete a`

LolfoollorS committed:
Refactored Storage.java

LolfoollorS committed:
Add Level-10 Gui

LolfoollorS committed:
no message

LolfoollorS committed:
Add C-Undo for undo personal

LolfoollorS committed:
Add A-UserGuide

LolfoollorS committed:
Add HelpWindow.fxml
Add HelpCommand.java
Adhere to Coding Standards

Change to assist people that requires help.
Add HelpCommand if user needs help.

LolfoollorS committed:
Add UI for Docs

LolfoollorS committed:
Change mainclassname in build.gradle

LolfoollorS committed:
Modify README.md

LolfoollorS committed:
Renam UI to Ui.png

LolfoollorS committed:
Improve dialog box

LolfoollorS committed:
Update CSS and FXML files

LolfoollorS committed:
Force Java 11 Compile.
Update Sprite for bot.

LolfoollorS committed:
Update app scene and Ui.png
